pkg/admission/mutator/shoot.go

```diff
@@ -89,70 +86,37 @@ func (s *shoot) Mutate(_ context.Context, newObj, oldObj client.Object) error {
 	overlayDisabled := false
 
 	if shoot.Spec.Networking != nil && shoot.Spec.Networking.Type != nil {
-		switch *shoot.Spec.Networking.Type {
-		case calico.ReleaseName:
-			overlay := &calicov1alpha1.Overlay{Enabled: false}
-
-			networkConfig, err := s.decodeCalicoNetworkConfig(shoot.Spec.Networking.ProviderConfig)
-			if err != nil {
-				return err
-			}
-
-			if oldShoot == nil && networkConfig.Overlay == nil {
-				networkConfig.Overlay = overlay
-			}
-
-			if oldShoot != nil && networkConfig.Overlay == nil {
-				oldNetworkConfig, err := s.decodeCalicoNetworkConfig(oldShoot.Spec.Networking.ProviderConfig)
-				if err != nil {
-					return err
-				}
-
-				if oldNetworkConfig.Overlay != nil {
-					networkConfig.Overlay = oldNetworkConfig.Overlay
-				}
-
-			}
-
-			if networkConfig.Overlay != nil && !networkConfig.Overlay.Enabled {
-				overlayDisabled = true
-			}
-
-			shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
-				Object: networkConfig,
-			}
+		networkConfig, err := s.decodeNetworkConfig(shoot.Spec.Networking.ProviderConfig)
+		if err != nil {
+			return err
+		}
 
-		case cilium.ReleaseName:
-			overlay := &ciliumv1alpha1.Overlay{Enabled: false}
```

Comment:
Do we want to use constants for `overlay` and `enabled` to prevent typos? These strings are used multiple times here.